Performance-Tracing.md

```diff
@@ -0,0 +1,156 @@
+
+5.  Navigate to [edge://tracing](edge://tracing) (or [chrome://tracing](chrome://tracing)) and click `Load`.\
+    Navigate to and select `trace.json`.
+
+6.  If you've never visited [edge://tracing](edge://tracing) before, you'll see something like this:\
+    ![Initial view of edge://tracing](https://raw.githubusercontent.com/wiki/Microsoft/TypeScript/images/tracingJustOpened.png)
+    1.  Use the little triangle to expand the view of the "Main" thread.
+        For convenience, we ignore all other threads.
+    2.  This little toolbar is floating - you'll probably want to move it into the blank space under the thread name (i.e. into the left sidebar).
+        Each button indicates a cursor mode.
+    3.  This mode is for selection.
+        Selecting a single element is possible in most modes, but box selection is only possible in this one.\
+        *Tip:* box selection is a great way to get a summary of a region.
+    4.  This mode is for panning.
+        Click-drag to scroll to other parts of the trace..
+    5.  This mode is for zooming.
+        Click-drag up or right to zoom in and down or left to zoom out.
+
+7.  If you expand the Main thread and select one of the boxes, you'll see something like this, with time on the horizontal axis and (partial) call stacks growing down from the top:\
+    ![Expanded view of edge://tracing](https://raw.githubusercontent.com/wiki/Microsoft/TypeScript/images/tracingExpanded.png)
+    1.  You can easily differentiate the four major phases of compilation.
+        This is program construction, which includes parsing and modules resolution.
+        Usually, you want to look for the largest `findSourceFile` boxes and follow them down to see which dependency chain is taking the most time.
+        Usually, it will be a library outside your own code.
+        Be sure to include this library in any repro you construct.
+    2.  This is the binding phase.
+        It is rarely an issue, so we'll ignore it for now.
+    3.  This is the (type) checking phase, where most of the TypeScript-specific work happens.
+        *You generally want to start here.*
+        We'll zoom in in a later step.
+    4.  This is the emit phase, which we'll ignore for now.
+    5.  This is the selected box.
+    6.  These are the built-in metadata about the box.
+        1.  `Start` is the timestamp of the start of the operation - usually uninteresting.
+        2.  `Wall Duration` this is what would be called "Total Time" in many profiling tools.
+        3.  `Self Time` is the amount of time that is not accounted for by child boxes (i.e. directly below this one).
+    7.  This metadata is specific to the kind of box you're looking at.
+        *This is the most important information* - it will contain paths and things.
+
+    *Note:* All times reflect the overhead of tracing, which is not spread uniformly through the execution.
```

Comment:
Drop "All", and maybe mention that it will hopefully be reduced to not cause visible noise?

Reply:
I like the "all" but I've added a note about future improvements.